cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -447,20 +446,43 @@ private void writeEntriesToContainerMetadata(DebugStreamSegmentContainer contain
                 tableExtension.remove(tableSegment, Collections.singletonList(unversionedEntry.getKey()), TIMEOUT);
             }
         }
+        //reset the core attribs in storage_segment so as to allow the attributes
+        //index to get generated with flush-to-storage being invoked.
+        resetStorageSegment(container);
     }
 
     /**
-     *  Resets the Storage segment
+     * Reset the storage_metadata segment,especially, some of the core attributes
+     * from the recovered segment. Resetting these will allow the operations
+     * parsed from storage_metadata chunks to be processed.
+     * @param container Container being recoverd.
+     * @throws Exception
      */
-    private MetadataStore.SegmentInfo resetStorageSegment(MetadataStore.SegmentInfo segmentInfo) {
+    private void resetStorageSegment(DebugStreamSegmentContainer container) throws Exception {
+        ContainerTableExtension tableExtension = container.getExtension(ContainerTableExtension.class);
+        List<TableEntry> entries = tableExtension.get(NameUtils.getMetadataSegmentName(container.getId()),
+                Collections.singletonList(BufferView.wrap(NameUtils.getStorageMetadataSegmentName(container.getId()).getBytes(StandardCharsets.UTF_8))), TIMEOUT).get();
+        TableEntry entry = entries.get(0);
+        TableEntry unversionedEntry = TableEntry.unversioned(entry.getKey().getKey(), entry.getValue());
+        MetadataStore.SegmentInfo segmentInfo = SERIALIZER.deserialize(new ByteArraySegment(unversionedEntry.getValue().getCopy()).getReader());
+
+        Map<AttributeId, Long> attribs = new HashMap<>(segmentInfo.getProperties().getAttributes());
```

Comment:
using `ImmutableMap` a good idea here? (it's a private method so I guess it's safe)